### PR TITLE
1887 [WIP] feature federation 09 create database and typeorm model

### PR DESCRIPTION
**Pull request by clauspeterhuebner**
### Issues

- None

### Todo

- [x] backend-config for DB version check is open
- [ ] linting of database creates a lot of changes, so don't dare to commit
- [ ] can't find an existing relation OneToMany <-> ManyToOne as template, so a new solution is created, pleasse check this

### CODE REVIEW

File: database/migrations/0037-community_federation_tables.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`community\` (
+      \`id\` int unsigned NOT NULL AUTO_INCREMENT,
+      \`uuid\` varchar(37) NOT NULL,
```
Comment (Mogge) [inline database/migrations/0037-community_federation_tables.ts:13]: I found this one on MariaDB knowledge:
```
UUID() results are intended to be unique, but cannot always be relied upon to unpredictable and unguessable, so should not be relied upon for these purposes.
```

Comment (Mogge) [reply database/migrations/0037-community_federation_tables.ts:13]: https://mariadb.com/kb/en/uuid/

File: database/migrations/0037-community_federation_tables.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      \`createdAt\` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`uuid\` (\`uuid\`)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+  `)
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`community_federation\` (
+      \`id\` int unsigned NOT NULL AUTO_INCREMENT,
+      \`uuid\` varchar(36) NOT NULL,
+      \`foreign\` tinyint DEFAULT '0',
```
Comment (Mogge) [inline database/migrations/0037-community_federation_tables.ts:25]: As stated above. When `true` I understand to have a foreign community. Change to `self` (if not reserved)

**Comment by clauspeterhuebner** [comment]
this PR is deprecated, replaced by #2339 and will be closed now

**Pull request closed without merging**